### ISSUE TITLE
#define AES_KEYLEN bites with SAMD5 compiler

### ISSUE THREAD

**Issue by vogt31337** [kokke/tiny-AES-c#125]
Hi,

your tiny AES works flawlessly on an Arduino Feather M4 Express (AVR SAMD5 architecture). 
But the #define AES_KEYLEN get's redefined in the include files from the compiler. The SAMD5 has a hardware AES module and defines it's own AES_KEYLEN. 

So maybe rename some things in this library?

BR

**Comment by DamonHD** [comment]
This would be the sort of thing that namespaces and static constexpr are designed to gracefully sidestep...

Rgds

Damon

**Comment by vogt31337** [comment]
I don't know maybe. But when I include tiny-AES-c the compiler complains. And I thought changing the compiler delivered includes wouldn't be the right spot...

**Comment by kokke** [comment]
Hi @vogt31337 and thanks for your interest in this project.

As I usually do, I agree with @DamonHD on this point.

> So maybe rename some things in this library?

I'll be happy to review a PR with any naming changes you could have in mind :)

**Comment by vogt31337** [comment]
I also agree with @DamonHD, but I don't have enough time / knowledge to implement this.
So as a matter of fact I would just change the name to something like TINY_AES_KEYLEN...
